### PR TITLE
154-Change repositoryImpl to spring data just for Class entity

### PR DESCRIPTION
**Pull request by mohsenmansourvar**

### CODE REVIEW

File: src/main/java/com/education/EducationApplication.java [hunk 1]
```diff
@@ -3,8 +3,12 @@
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
 
+@EnableJpaRepositories
 @SpringBootApplication(exclude = HibernateJpaAutoConfiguration.class)
+@EnableTransactionManagement
```
Comment (mohsenebrahimi123) [inline src/main/java/com/education/EducationApplication.java]: Sort annotations by its length:
@AnAnnotation
@ALongAnnotation
@AVeryVeryLongAnnotation

Comment (mohsenebrahimi123) [inline src/main/java/com/education/EducationApplication.java]: Remove EnableTransactionManagement annotation from `EducationConfiguration` class:
https://github.com/mohsenmansourvar/education/blob/bd32886c7e1a164839e38be4ca4259a9521eb71c/src/main/java/com/education/configuration/EducationConfiguration.java#L17

File: src/main/java/com/education/repository/ClassRepositoryImpl.java [hunk 2]
```diff
@@ -57,3 +58,4 @@ public void update(long id, Class newClass) {
                 .list();
     }
 }
+*/
```
Comment (mohsenebrahimi123) [inline src/main/java/com/education/repository/ClassRepositoryImpl.java]: Why don't we delete the class?

File: src/main/java/com/education/service/ClassService.java [hunk 3]
```diff
@@ -5,13 +5,16 @@
...
     void delete(long id);
 
-    void update(long id, Class newRoom);
+    Class getById(long id);
+
+    List<Class> getAllClasses();
 
-    List<Class> getAllRooms();
 }
+
```
Comment (mohsenebrahimi123) [inline src/main/java/com/education/service/ClassService.java:20]: One space is enough at the end of the class

File: src/main/java/com/education/service/ClassService.java [hunk 4]
```diff
@@ -5,13 +5,16 @@
 import java.util.List;
 
 public interface ClassService {
+
     void save(Class aClass);
```
Comment (mohsenebrahimi123) [inline src/main/java/com/education/service/ClassService.java]: The return type of save method should be Class

File: src/main/java/com/education/service/ClassServiceImpl.java [hunk 5]
```diff
@@ -2,38 +2,49 @@
...
-    private final ClassRepository classRepository;
 
-    public ClassServiceImpl(ClassRepository classRepository) {
-        this.classRepository = classRepository;
-    }
+    private final ClassRepository classRepository;
 
     @Override
+    @Transactional
     public void save(Class aClass) {
```
Comment (mohsenebrahimi123) [inline src/main/java/com/education/service/ClassServiceImpl.java]: The return type of save method should be Class

File: src/main/java/com/education/service/ClassServiceImpl.java [hunk 6]
```diff
@@ -2,38 +2,49 @@
...
         classRepository.save(aClass);
```
Comment (mohsenebrahimi123) [inline src/main/java/com/education/service/ClassServiceImpl.java]: replace this with:
```
return classRepository.save(aClass);
```

File: src/test/java/com/education/service/ClassServiceTest.java [hunk 7]
```diff
@@ -4,97 +4,94 @@
...
     @Test
     public void save() {
-        Class room = new Class();
-        room.setClassNumber("125");
-        room.setCapacity(45);
-        classService.save(room);
-        Long roomId = room.getId();
+        Class aClass = new Class();
+        aClass.setClassNumber("125");
+        aClass.setCapacity(45);
```
Comment (mohsenebrahimi123) [inline src/test/java/com/education/service/ClassServiceTest.java]: Duplicated code. Create a method in this class that accepts classNumber and capacity and creates a class.

Comment (mohsenebrahimi123) [reply src/test/java/com/education/service/ClassServiceTest.java]: Use Builder method

File: src/test/java/com/education/service/ClassServiceTest.java [hunk 8]
```diff
@@ -4,97 +4,94 @@
...
+        classService.save(aClass);
 
-        Class roomById = classService.getById(roomId);
+        Class ClassById = classService.getById(aClass.getId());
```
Comment (mohsenebrahimi123) [inline src/test/java/com/education/service/ClassServiceTest.java:28]: Not needed. Just use the return value of classService.save method. It returns an object with id

Comment (mohsenebrahimi123) [reply src/test/java/com/education/service/ClassServiceTest.java:28]: `ClassById` -> `classById`

File: src/test/java/com/education/service/ClassServiceTest.java [hunk 9]
```diff
@@ -4,97 +4,88 @@
...
     }
 
     @Test
     public void delete() {
-        Class room = new Class();
-        room.setClassNumber("111");
-        room.setCapacity(20);
-        classService.save(room);
-        Long roomId = room.getId();
+        Class class1 = SampleBuilder.class1();
```
Comment (mohsenebrahimi123) [inline src/test/java/com/education/service/ClassServiceTest.java:49]: why class1?
I think 1 is misleading.
Also, method name should be verb

Comment (mohsenmansourvar) [reply src/test/java/com/education/service/ClassServiceTest.java:49]: can I consider createClass1?

**Review by mohsenebrahimi123 — APPROVED** [review]

**Pull request merged**